File: examples/matrix/Dockerfile.java
# Java runtime column.
FROM eclipse-temurin:21-jdk
COPY adapters/java /src/java
WORKDIR /src/java
CMD ["bash","-c","d=$(mktemp -d); javac -d \"$d\" Urirun.java example/HashConnector.java && java -cp \"$d\" HashConnector > /shared/java.json && echo 'java: emitted /shared/java.json'"]

File: examples/matrix/Dockerfile.go
# Go runtime column: emit the shared hash bindings via the Go SDK example.
FROM golang:1.22-bookworm
COPY adapters/go /src/go
WORKDIR /src/go
# Write the emitted contract to the shared volume for the matrix verifier.
CMD ["sh", "-c", "go run ./example/hash-connector > /shared/go.json && echo 'go: emitted /shared/go.json'"]
